agent/connect/ca/provider_vault_test.go

```diff
@@ -924,6 +924,76 @@ func TestVaultCAProvider_GenerateIntermediate(t *testing.T) {
 	require.NotEqual(t, orig, new)
 }
 
+func TestVaultCAProvider_GenerateIntermediate_inSecondary(t *testing.T) {
```

Comment:
This is a mirror of the test added in https://github.com/hashicorp/consul/pull/15253 to validate the primary side of the bug.